### PR TITLE
Some code smell fixes

### PR DESCRIPTION
**Pull request by mattch1**
Addressed 3/8 code smells. Apparently adding comments to @Ignore should stop Sonar complaining about these.

The other 5 I'm not sure about they may break the code as Sonar itself is not taking into account what the code is being used in conjunction with; they can be seen here:

http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aextensions-api&resolved=false&types=CODE_SMELL

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java [hunk 1]
```diff
@@ -118,8 +119,9 @@ public FileTransferApiClientResponse download(String fileId, HttpServletResponse
      * @param clientHttpResponse the response back from the api we are calling - the file-transfer-api
      */
     private void setResponseHeaders(HttpServletResponse httpServletResponse, ClientHttpResponse clientHttpResponse) {
-        HttpHeaders incomingHeaders = clientHttpResponse.getHeaders();
-        if (incomingHeaders != null) {
+        Optional<HttpHeaders> incomingHeadersOpt = Optional.of(clientHttpResponse.getHeaders());
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java:122]: Sonar threw a failure on this new change, looks like the same reason as the old version.  clientHttpResponse.getHeaders() cannot return null so the fix would be just to have the previous code and remove "if (incomingHeaders != null)"

File: src/test/java/uk/gov/companieshouse/extensions/api/requests/MongoDBTest.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public void removeTestDocuments() {
         requestsRepository.deleteById(SUMMER_TEST_DOCUMENT);
     }
 
-    @Ignore
+    @Ignore("This requires an in-memory db similar to Derby but for Mongo")
```
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/extensions/api/requests/MongoDBTest.java:80]: looks like these are ignored as we need a mongoDB connection/db to run these and concourse doesn't have one. I was thinking, as this test is marked with a category of "integration", concourse will try and execute it. Perhaps for a future Jira story, we could look into having a separate category for this and other DB tests that we could add into the makefile for running locally only as we'll forget that we have to un-ignore them.

**Comment by markpit** [comment]
Typo in commit message

**Comment by mattch1** [comment]
Any thoughts about the other 5 code smells? - does anyone think they can be addressed?

**Comment by ICoxCH** [comment]
> Any thoughts about the other 5 code smells? - does anyone think they can be addressed?

This smell http://code-analysis.platform.aws.chdev.org:9000/project/issues?id=uk.gov.companieshouse%3Aextensions-api&open=AW1Ee9gEu7PxO9thJPwV&resolved=false&types=CODE_SMELL
Can we remove the Mappings annotation? And just have the `@Mapping` annotation? Sonar seems to think that would be fine.

**Comment by mattch1** [comment]
I suspect the Sonar message is very generic but because of something specific about the mappings annoation the attachments functionality stopped working when I tried to run this with out it. I got the "problem with the service" page plus the attachments were missing in mongo.

File: src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java]: Addressed this.

File: src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Collections;
+import java.util.Optional;
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java:6]: optional is an unused import

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java]: Gone.

**Pull request merged**